include/matrix_assembler.cc

```diff
@@ -0,0 +1,151 @@
+// Constructor
+MatrixAssembler::MatrixAssembler() {
+  jac_ = std::make_shared<Eigen::SparseMatrix<double>>();
+  node_head_vec_ = std::make_shared<Eigen::VectorXd>();
+  node_discharge_vec_ = std::make_shared<Eigen::VectorXd>();
+  pipe_discharge_vec_ = std::make_shared<Eigen::VectorXd>();
+};
+
+// Assign global index to nodes in the whole network
+void MatrixAssembler::assign_global_nodal_indices(
+    std::vector<std::shared_ptr<pipenetwork::Node>> nodes) {
+  global_nodes_.clear();
+  Index index = 0;
+  for (const auto& node : nodes) {
+    global_nodes_.emplace(
+        std::pair<std::shared_ptr<pipenetwork::Node>, Index>(node, index));
+    ++index;
+  }
+  nnode_ = global_nodes_.size();
+}
+
+// Assign global index to pipes in the whole network
+void MatrixAssembler::assign_global_pipe_indices(
```

Comment:
If the mesh class has all the pipes, then do we need this global index? Can we not use the index from the mesh class itself?

Reply:
Same as above, is it possible that different parts of the network are in different meshes?

Reply:
No, let's just have numbering at the mesh class level